gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/javaversions/BaselineJavaVersion.java

```diff
@@ -198,6 +199,13 @@ public void execute(Test test) {
                 test.getJvmArgumentProviders().add(new EnablePreviewArgumentProvider(runtime));
             }
         });
+
+        project.getTasks().withType(Checkstyle.class).configureEach(new Action<Checkstyle>() {
+            @Override
+            public void execute(Checkstyle checkstyle) {
+                checkstyle.getJavaLauncher().set(javaToolchain.flatMap(BaselineJavaToolchain::javaLauncher));
```

Comment:
I think we want to use the compile target for checkstyle, not the runtime java version

Reply:
you are right. Moved into the compile section